### ISSUE TITLE
Stuck in imagemin loop

### ISSUE THREAD

**Issue by BrizzleRocker** [fourkitchens/emulsify-gulp#82]
Node 8.11.3
NPM 5.6

Running in a docker container based on node:8.11

I run `npm install --unsafe-perm` and then `npm start`

I then get the following:
```
[13:45:03] Finished 'css' after 353 ms
[13:45:03] Starting 'serve'...
[13:45:04] Finished 'serve' after 1.18 s
[13:45:04] Starting 'theme'...
[13:45:04] Finished 'theme' after 34 μs
[Browsersync] Access URLs:
 -----------------------------------------------------
    Local: http://localhost:3000/pattern-lab/public/
 External: http://172.19.0.10:3000/pattern-lab/public/
 -----------------------------------------------------
[Browsersync] Serving files from: ./
[13:45:04] Starting 'imagemin'...
[13:45:04] Finished 'imagemin' after 1.29 ms
[13:45:05] Starting 'imagemin'...
[13:45:05] Finished 'imagemin' after 710 μs
[13:45:05] Starting 'imagemin'...
[13:45:05] Finished 'imagemin' after 592 μs
[13:45:05] Starting 'imagemin'...
[13:45:05] Finished 'imagemin' after 2.26 ms
[13:45:05] Starting 'imagemin'...
[13:45:05] Finished 'imagemin' after 694 μs
[13:45:05] Starting 'imagemin'...
[13:45:05] Finished 'imagemin' after 705 μs
[13:45:05] Starting 'imagemin'...
[13:45:05] Finished 'imagemin' after 643 μs
```
This just keeps going, imagemin never finishes. Happens if I try and run `npm start` outside of the container too.

Would appreciate some help as failing PL is now stopping the entire dev team.

**Comment by BrizzleRocker** [comment]
Ref: https://github.com/JeremyEnglert/JointsWP/issues/300

**Comment by BrizzleRocker** [comment]
More follow-up as my colleague @bash247 is having the same issue.

Re-running `npm install --unsafe-perm` and then `npm install singularitygs --save-dev` was his initial fix but I saw an error with missing emulsify-gulp so ran `npm install emulsify-gulp` and it now runs fine.

Odd.

**Comment by bash247** [comment]
@BrizzleRocker and I have made further progress on this issues. After re-running, `npm install --unsafe-perm`, `npm install singularitygs --save-dev` and then `npm install emulsify-gulp`. We can clean, build and generate the site by running the tasks individually by running `./node_modules/.bin/gulp (pl:full) (build) (clean) (theme)` - remove the brackets before running commands. 

Stepping into the code to identify point of failure, it appears to throw an uncaught error on gulp-svg-sprite. 

Command which was run - `node --inspect-brk ./node_modules/gulp/bin/gulp.js build --verbose`

**Comment by BrizzleRocker** [comment]
@bash247 https://github.com/jkphl/gulp-svg-sprite/issues/82

**Comment by BrizzleRocker** [comment]
```
> emulsify@1.0.0 start /var/www/focusrite_develop_build_24/docroot/sites/focusrite/themes/custom/PatternLabFocusrite_emulsify
> gulp clean && gulp compile && gulp theme

[16:49:40] Using gulpfile /var/www/focusrite_develop_build_24/docroot/sites/focusrite/themes/custom/PatternLabFocusrite_emulsify/gulpfile.js
[16:49:40] Starting 'clean:css'...
[16:49:40] Finished 'clean:css' after 5.05 ms
[16:49:40] Starting 'clean'...
[16:49:40] Finished 'clean' after 26 μs
[16:49:42] Using gulpfile /var/www/focusrite_develop_build_24/docroot/sites/focusrite/themes/custom/PatternLabFocusrite_emulsify/gulpfile.js
[16:49:42] Starting 'icons'...
[16:49:42] Finished 'icons' after 14 ms
[16:49:42] Starting 'pl:scss-to-yaml'...
[16:49:42] Finished 'pl:scss-to-yaml' after 4.7 ms
[16:49:42] Starting 'pl:full'...
configuring pattern lab...
you may have a typo in 04-templates/example/example.twig. `organisms-info-box-grid` is not a valid pattern...
you may have a typo in 05-pages/example/example.twig. `organisms-info-box-grid` is not a valid pattern...
your site has been generated...
site generation took 0.83149981498718 seconds and used 24MB of memory...
[16:49:43] Finished 'pl:full' after 886 ms
[16:49:43] Starting 'compile'...
[16:49:43] Finished 'compile' after 26 μs
[16:49:46] Using gulpfile /var/www/focusrite_develop_build_24/docroot/sites/focusrite/themes/custom/PatternLabFocusrite_emulsify/gulpfile.js
[16:49:46] Starting 'imagemin'...
[16:49:46] Finished 'imagemin' after 59 ms
[16:49:46] Starting 'css'...
[16:49:46] Starting 'scripts'...
[16:49:46] Finished 'scripts' after 881 μs
[16:49:46] Starting 'styleguide-scripts'...
[16:49:46] Finished 'styleguide-scripts' after 946 μs
[16:49:46] Starting 'watch:pl'...
[16:49:47] Finished 'watch:pl' after 256 ms
DEPRECATION WARNING: Passing a string to call() is deprecated and will be illegal
in Sass 4.0. Use call(get-function("output-isolation")) instead.

DEPRECATION WARNING: Passing a string to call() is deprecated and will be illegal
in Sass 4.0. Use call(get-function("output-isolation")) instead.

[16:49:51] gulp-imagemin: Minified 55 images (saved 63.7 kB - 29.9%)
[16:49:51] Finished 'css' after 5.1 s
[16:49:51] Starting 'serve'...
[16:49:53] Finished 'serve' after 1.82 s
[16:49:53] Starting 'theme'...
[16:49:53] Finished 'theme' after 49 μs
[Browsersync] Access URLs:
 -----------------------------------------------------
    Local: http://localhost:3000/pattern-lab/public/
 External: http://172.30.1.21:3000/pattern-lab/public/
 -----------------------------------------------------
[Browsersync] Serving files from: ./
[16:49:53] Starting 'imagemin'...
[16:49:53] Finished 'imagemin' after 1.71 ms
[16:49:53] Starting 'imagemin'...
[16:49:53] Finished 'imagemin' after 1.07 ms
[16:49:53] Starting 'imagemin'...
[16:49:53] Finished 'imagemin' after 1.01 ms
```

Despite minifying in the compile task it insists on doing them again. (In order to get this far I had to follow the steps in https://github.com/fourkitchens/emulsify-gulp/issues/82#issuecomment-402987711 above

**Comment by kiwimind** [comment]
Amending the `node_modules/emulsify-gulp/index.js` file to stop it calling imagemin is about the only thing we can do at the moment to keep things working as expected.

**Comment by evanmwillhite** [comment]
We've made some tweaks to where imagemin is run. When I merged it automatically closed this issue, but I've reopened. Can I get some confirmation that this has solved this issue? Thank you all.

**Comment by kiwimind** [comment]
@evanmwillhite Thanks for that. I can confirm that it's working fine (once I'd removed the `gulp clean` from the package.json too 😄 ).

**Comment by kiwimind** [comment]
Any chance there's a release coming any time soon? ;)

**Comment by evanmwillhite** [comment]
Awesome! Thanks @kiwimind. I'll see if we can get one out today.

**Comment by evanmwillhite** [comment]
@kiwimind just a heads-up that we released 2.7 today which includes this. Thanks!

**Comment by kiwimind** [comment]
Ah cool, nice one. Thanks for letting me know.

We're currently pinning to a commit, so will look at using the new version to see how that goes.
